cognite/client/data_classes/geospatial.py

```diff
@@ -357,3 +358,65 @@ def _load(cls, resource: Dict, cognite_client: "CogniteClient" = None) -> "Raste
             snake_case_key = to_snake_case(key)
             setattr(instance, snake_case_key, value)
         return instance
+
+
+class GeospatialComputeFunction(ABC):
+    """A geospatial compute function"""
+
+    @abstractmethod
+    def to_json_payload(self) -> dict:
+        """Convert function to json for request payload"""
+
+
+class GeospatialGeometryTransformComputeFunction(GeospatialComputeFunction):
+    "A stTransform geospatial compute function"
+
+    def __init__(self, geospatial_geometry_compute_function: GeospatialComputeFunction, srid: int):
+        self.geospatial_geometry_compute_function = geospatial_geometry_compute_function
+        self.srid = srid
+
+    def to_json_payload(self) -> dict:
+        return {"geometry": self.geospatial_geometry_compute_function.to_json_payload(), "srid": self.srid}
+
+
+class GeospatialGeometryComputeFunction(GeospatialComputeFunction, ABC):
```

Comment:
I'm a bit worried about strong typing all the functions, in terms of being a maintenance nightmare. Although technically I cannot disagree this has some advantages...

Reply:
I figured this might be an easier way to approach the way the functions seem to be able to nest within one another for geospatial. Mainly stole the approach from the seismic sdk https://github.com/cognitedata/cognite-seismic-sdk/blob/master/python/cognite/seismic/data_classes/searchspec.py we was implemented to move away from our client's search functions from having 100 arguments 😅